vpr/src/base/FlatPlacementInfo.h

```diff
@@ -0,0 +1,71 @@
+/**
+ * @file
+ * @author  Alex Singer
+ * @date    January 2025
+ * @brief   Declaration of the FlatPlacementInfo object, which is used to store
+ *          flat placement information used by the packer.
+ */
+
+#pragma once
+
+#include "atom_netlist.h"
+#include "vtr_vector.h"
+
+/**
+ * @brief Flat placement storage class.
+ *
+ * This stores placement information for each atom in the netlist. It contains
+ * any information that may be used by the packer to better create clusters.
+ */
+class FlatPlacementInfo {
+public:
+    /// @brief Identifier for an undefined position.
+    static constexpr float UNDEFINED_POS = -1.f;
+    /// @brief Identifier for an undefined sub tile.
+    static constexpr int UNDEFINED_SUB_TILE = -1;
+    /// @brief Identifier for an undefined site idx.
+    static constexpr int UNDEFINED_SITE_IDX = -1;
+
+    /// @brief The x-positions of each atom block. Is UNDEFINED_POS if undefined.
+    vtr::vector<AtomBlockId, float> blk_x_pos;
+    /// @brief The y-positions of each atom block. Is UNDEFINED_POS if undefined.
+    vtr::vector<AtomBlockId, float> blk_y_pos;
+    /// @brief The sub tile location of each atom block. Is UNDEFINED_SUB_TILE
+    ///        if undefined.
+    vtr::vector<AtomBlockId, int> blk_sub_tile;
+    /// @brief The flat site idx of each atom block. Is UNDEFINED_SITE_IDX if
```

Comment:
Explain what a flat_site_idx is -- an optional index into a linearized list of primitive locations within a cluster-level block.

Also, should add layer.